arch/xtensa/src/esp32/esp32_ble_adapter.c

```diff
@@ -312,47 +312,49 @@ struct irqstate_list_s
  */
 
 static xt_handler ble_set_isr(int n, xt_handler f, void *arg);
-static void ints_on(uint32_t mask);
+static void ints_on(unsigned int mask);
 static void IRAM_ATTR interrupt_disable(void);
 static void IRAM_ATTR interrupt_restore(void);
 static void IRAM_ATTR task_yield_from_isr(void);
 static void *semphr_create_wrapper(uint32_t max, uint32_t init);
 static void semphr_delete_wrapper(void *semphr);
-static int IRAM_ATTR semphr_take_from_isr_wrapper(void *semphr, void *hptw);
-static int IRAM_ATTR semphr_give_from_isr_wrapper(void *semphr, void *hptw);
-static int semphr_take_wrapper(void *semphr, uint32_t block_time_ms);
-static int semphr_give_wrapper(void *semphr);
+static int32_t IRAM_ATTR semphr_take_from_isr_wrapper(void *semphr,
+                                                      void *hptw);
+static int32_t IRAM_ATTR semphr_give_from_isr_wrapper(void *semphr,
+                                                      void *hptw);
+static int32_t semphr_take_wrapper(void *semphr, uint32_t block_time_ms);
+static int32_t semphr_give_wrapper(void *semphr);
 static void *mutex_create_wrapper(void);
 static void mutex_delete_wrapper(void *mutex);
-static int mutex_lock_wrapper(void *mutex);
-static int mutex_unlock_wrapper(void *mutex);
+static int32_t mutex_lock_wrapper(void *mutex);
+static int32_t mutex_unlock_wrapper(void *mutex);
 static void *queue_create_wrapper(uint32_t queue_len, uint32_t item_size);
 static void queue_delete_wrapper(void *queue);
-static int queue_send_wrapper(void *queue,
+static int32_t queue_send_wrapper(void *queue,
                               void *item,
                               uint32_t block_time_ms);
-static int IRAM_ATTR queue_send_from_isr_wrapper(void *queue,
-                                                 void *item,
-                                                 void *hptw);
-static int queue_recv_wrapper(void *queue,
-                              void *item,
-                              uint32_t block_time_ms);
-static int IRAM_ATTR queue_recv_from_isr_wrapper(void *queue,
-                                                 void *item,
-                                                 void *hptw);
-static int task_create_wrapper(void *task_func,
-                               const char *name,
-                               uint32_t stack_depth,
-                               void *param,
-                               uint32_t prio,
-                               void *task_handle,
-                               uint32_t core_id);
+static int32_t IRAM_ATTR queue_send_from_isr_wrapper(void *queue,
+                                                     void *item,
+                                                     void *hptw);
+static int32_t queue_recv_wrapper(void *queue,
```

Comment:
just a curious question: what's the policy on int32_t vs int in esp32 code?

even the "upstream" (esp-hal-3rdparty) seems inconsistent to me.

Reply:
`esp-hal-3rdparty` "follows" ESP-IDF changes and Espressif is continuously updating it to use the portable options. The same is happening with `esp-hal-3rdparty`.

You can refer to https://docs.espressif.com/projects/esp-idf/en/latest/esp32/migration-guides/release-5.x/5.0/gcc.html#espressif-toolchain-changes for more information.